File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0;
        unordered_map<int,int>mpp;
        int curSum = 0;
        mpp[curSum] = 1;
        
        for(int i=0;i<nums.size();i++)
        {
            curSum += nums[i];
            int dif = curSum - k;
            if(mpp.find(dif) != mpp.end())
            {
                count += mpp[dif];
            }
            mpp[curSum]++;

        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix_count = defaultdict(int)
        prefix_count[0] = 1
        current_sum = 0
        res = 0

        for num in nums:
            current_sum += num
            res += prefix_count[current_sum-k]
            prefix_count[current_sum] += 1
        return res
